### PR TITLE
Add simple Registration-Process

### PR DESCRIPTION
**Pull request by Cathesso**
Frontend: Compare password with passwordChecked and check if password is valid.
Backend: Create new appUser and save in DB
FE + BE: Check if userName is already in use.

### CODE REVIEW

File: frontend/src/pages/RegisterPage.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            fullWidth={true}
+            name="username"
+            error={usernameAlreadyRegistered}
+            onChange={handleChange}
+            value={registerCredentials.username}
+          />
+
+          {usernameAlreadyRegistered && (
+            <FormHelperText variant="outlined" margin="dense">
+              Username already taken :C
```
Comment (yasminwaldeck) [inline frontend/src/pages/RegisterPage.js:100]: :C

File: frontend/src/pages/RegisterPage.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    ) {
+      register(registerCredentials);
+    } else {
+      alert("There is something wrong with your chosen Username or Password");
+    }
+  };
+
+  function checkPassword() {
+    const matcher =
+      /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,50}$/;
```
Comment (yasminwaldeck) [inline frontend/src/pages/RegisterPage.js:74]: uiuiuiuiuiui

**Review by yasminwaldeck — APPROVED** [review]
huiuiui

File: frontend/src/pages/RegisterPage.js (hunk 1 above)
Comment (Cathesso) [reply frontend/src/pages/RegisterPage.js:100]: :D

File: frontend/src/pages/RegisterPage.js (hunk 2 above)
Comment (Cathesso) [reply frontend/src/pages/RegisterPage.js:74]: 🤓

**Pull request merged**